// react-mdl/lib/Grid.js
'use strict';

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.Cell = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var Grid = function Grid(props) {
    var noSpacing = props.noSpacing;
    var className = props.className;
    var children = props.children;

    var otherProps = _objectWithoutProperties(props, ['noSpacing', 'className', 'children']);

    var classes = (0, _classnames2.default)('mdl-grid', {
        'mdl-grid--no-spacing': noSpacing
    }, className);

    return _react2.default.createElement(
        'div',
        _extends({ className: classes }, otherProps),
        children
    );
};

Grid.propTypes = {
    className: _react.PropTypes.string,
    noSpacing: _react.PropTypes.bool
};

/* eslint-disable react/no-multi-comp */
var Cell = function Cell(props) {
    var _classNames;

    var align = props.align;
    var className = props.className;
    var children = props.children;
    var col = props.col;
    var phone = props.phone;
    var tablet = props.tablet;

    var otherProps = _objectWithoutProperties(props, ['align', 'className', 'children', 'col', 'phone', 'tablet']);

    var classes = (0, _classnames2.default)('mdl-cell', (_classNames = {}, _defineProperty(_classNames, 'mdl-cell--' + col + '-col', true), _defineProperty(_classNames, 'mdl-cell--' + phone + '-col-phone', typeof phone !== 'undefined'), _defineProperty(_classNames, 'mdl-cell--' + tablet + '-col-tablet', typeof tablet !== 'undefined'), _defineProperty(_classNames, 'mdl-cell--' + align, typeof align !== 'undefined'), _classNames), className);

    return _react2.default.createElement(
        'div',
        _extends({ className: classes }, otherProps),
        children
    );
};

Cell.propTypes = {
    align: _react.PropTypes.oneOf(['top', 'middle', 'bottom', 'stretch']),
    className: _react.PropTypes.string,
    col: _react.PropTypes.number.isRequired,
    phone: _react.PropTypes.number,
    tablet: _react.PropTypes.number
};

exports.default = Grid;
exports.Cell = Cell;
